pybamm/discretisations/discretisation.py

```diff
@@ -1197,3 +1200,85 @@ def check_variables(self, model):
                             var.shape, model.rhs[rhs_var].shape, var
                         )
                     )
+
+    def search_for_independent_var(self, model, var):
+        pybamm.logger.info("Removing independent blocks.")
+        boundary_variables = list(model.boundary_conditions.keys())
+        boundary_variable_keys = []
+        for condition in boundary_variables:
+            keys_for_condition = list(model.boundary_conditions[condition].keys())
+            boundary_variable_keys.append(keys_for_condition)
+        rhs_variables = list(model.rhs.keys())
+        algebraic_variables = list(model.algebraic.keys())
+        this_var_list = []
+        for child in var.children:
+            for tree in rhs_variables:
+                pybamm.tree_search(model.rhs[tree], child, this_var_list)
+            for tree in algebraic_variables:
+                pybamm.tree_search(model.algebraic[tree], child, this_var_list)
+            for (keys, tree) in zip(boundary_variable_keys, boundary_variables):
+                for key in keys:
+                    pybamm.tree_search(
+                        model.boundary_conditions[tree][key][0],
+                        child,
+                        this_var_list,
+                    )
+            for tree in model.variables.keys():
+                for rhs_child in model.variables[tree].children:
+                    pybamm.tree_search(rhs_child, var, this_var_list)
+        for tree in rhs_variables:
+            pybamm.tree_search(model.rhs[tree], var, this_var_list)
+        for tree in algebraic_variables:
+            pybamm.tree_search(model.algebraic[tree], var, this_var_list)
+        for (keys, tree) in zip(boundary_variable_keys, boundary_variables):
+            for key in keys:
+                pybamm.tree_search(
+                    model.boundary_conditions[tree][key][0], var, this_var_list
+                )
+        for tree in model.variables.keys():
```

Comment:
minor comment, `tree` here will be a string, not an expression tree, so `name` would be more appropriate